### ISSUE TITLE
Fix for issue #52 broke everything

### ISSUE THREAD

**Issue by Ribesg** [neuland/jade4j#147]
I've got a jade template extending a layout file placed in the parent folder:
- `kek/index.jade`
- `_layout.jade`

I'm using `extends ../_layout.jade`. It worked in `1.2.3`, but not in `1.2.4`. You might want to add a little test for this one.

I traced the issue down to `de.neuland.jade4j.template.ClasspathTemplateLoader`.
- Using version `1.2.3`, when calling `ClasspathTemplateLoader.getReader()`, the passed parameter is `jade\_layout.jade`*. This works when calling `getResourceAsStream`.
- Using version `1.2.4`, the passed parameter is now `jade\general/../_layout.jade`, which does not work. `getResourceAsStream` returns null.

*Yes, this is running in debug mode on my windows. I'm not sure if using `"/"` in Jade4J's code is a good idea, but it worked until now.

**Comment by chbloemer** [comment]
Please try 1.2.5. I added a basePath property to JadeConfiguration. This is used to resolve paths starting with "/" within the templates.
It is not possible to breakout of the template folder, so _layout.jade should be in the template folder.

For Example:
* /templates/
* /templates/pages
* /templates/pages/index.jade
* /templates/pages/include.jade
* /templates/_layout.jade

When "/templates" is your template folder and "pages" is your basePath, your index.jade could look like this:
```jade
extends ../_layout
include /include
``` 

* jade\general/../_layout.jade ist normalized to jade/_layout.jade now.
* ../_layout.jade can not be normalized because there is no parent folder.

Hope that helps.

**Comment by samogot** [comment]
@chbloemer, it isn't good enough for me. My case is:
- /app/
- /app/blocks/
- /app/blocks/layout-default/
- /app/blocks/layout-default/default-include.jade
- /app/pages/
- /app/pages/index.jade

index.jade looks like:
```
extends ../blocks/layout-default/layout-default
```

I'm using spring-jade4j with `spring.jade4j.prefix=classpath:/app/pages/`
so in [resolvePath](https://github.com/neuland/jade4j/blob/master/src/main/java/de/neuland/jade4j/parser/PathHelper.java#L7) I've got `parentName="index"`

as result:
```
de.neuland.jade4j.exceptions.JadeParserException: the template [../blocks/layout-default/layout-default.jade] could not be opened
class path resource [app/pages/null.jade] cannot be opened because it does not exist in index:1
```

**Comment by KennethNickles** [comment]
Is this resolved? I noticed 1.2.5 was released.

**Comment by fabiangr** [comment]
I am closing this issue as it has been resolved with 1.2.5 release. Please notice that no (included) templates may reside outside of the basePath when using a basePath or jade4j-spring.
